tests/driver_netdev_common/Kconfig

```diff
@@ -0,0 +1,375 @@
+# Copyright (c) 2021 Gunar Schorcht
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+
+config DRIVER_NETDEV_COMMON
+    bool
+    default y if TEST_KCONFIG
+
+    # Modules to include:
+    select MODULE_SHELL
+    select MODULE_SHELL_COMMANDS
+    select MODULE_PS
+
+    # gnrc is a meta module including all required, basic gnrc networking modules
+    select MODULE_GNRC
+
+    # automatically initialize the network interface
+    select MODULE_AUTO_INIT_GNRC_NETIF
+
+    # shell command to send L2 packets with a simple string
+    select MODULE_GNRC_TXTSND
+
+    # the application dumps received packets to stdout
+    select MODULE_GNRC_PKTDUMP
+
+    # TODO: parameters defined in default-radio-settings.inc.mk
+
+# the following is just for testing and is removed before merge
+
+if TEST_KCONFIG
+
+config MODULE_NETOPT
+    bool
+
+config MODULE_NETDEV
+    bool
+    default y if BOARD_ATXMEGA_A1_XPLAINED && MODULE_NETDEV_DEFAULT
+    default y if BOARD_ATXMEGA_A3BU_XPLAINED && MODULE_NETDEV_DEFAULT
+    select MODULE_EUI_PROVIDER
+
+config MODULE_NETDEV_ETH
+    bool
+    default y if HAS_PERIPH_ETH
+    select MODULE_NATIVE_CLI_EUI_PROVIDER if BOARD_NATIVE
+    select MODULE_NETDEV
+    select MODULE_NETDEV_REGISTER if !MODULE_GNRC_NETIF_SINGLE
+
+    select MODULE_STM32_ETH if HAS_PERIPH_ETH && HAS_CPU_STM32 && !MODULE_ATWINC15X0
+    select MODULE_SAM0_ETH if HAS_PERIPH_ETH && HAS_CPU_SAMD5X && !MODULE_ATWINC15X0
```

Comment:
I know this is here just for testing, but we can use it as a starting point.

As I understand the current modules, `MODULE_NETDEV_DEFAULT` will only enable networking devices that are declared by the board (via `HAVE_` symbols). Each driver would select the required netdev module, which would in turn select the generic netdev module. I don't think there is need to actually add prompts to these netdev modules, as they are needed.

Perhaps we need one more abstraction between driver module and netdev module, as some devices work without netdev (e.g., they use the IEEE802154 radio HAL). In that case the device would select instead a `HAVE_IEEE802154_DEVICE` or so, and the selection of the netdev may be optional if the driver supports something else than the netdev interface.

Reply:
Although highly outdated (both name schemes and some patterns), #14055 did some modelling in this direction, both netdev and IEEE802154-related GNRC modules. Note that the netdev modules had prompt in that proposal, but I'd not go with that now.